safe/gui/widgets/dock.py

```diff
@@ -1054,6 +1058,46 @@ def progress_callback(self, current_value, maximum_value, message=None):
         self.progress_bar.setValue(current_value)
         QtGui.QApplication.processEvents()
 
+    def generate_impact_report(self, impact_function):
+        """
+
+        :param impact_function:
+        :type impact_function: ImpactFunction
+        :return:
+        """
+        # get minimum needs profile
+        minimum_needs = NeedsProfile()
+        minimum_needs.load()
+
+        # create impact report instance
+        report_metadata = ReportMetadata(
+            metadata_dict=standard_impact_report_metadata)
+        impact_report = ImpactReportV4(
+            self.iface,
+            report_metadata,
+            impact_function=impact_function)
+
+        # generate report folder
+
+        # Set a unique name for this impact
+        _unique_name = impact_function.impact.name().replace(' ', '')
+        _unique_name = replace_accentuated_characters(_unique_name)
+        now = datetime.now()
+        date = now.strftime('%d%B%Y').decode('utf8')
+        time = now.strftime('%Hh%M-%S').decode('utf8')
+        _unique_name = '%s_%s_%s' % (_unique_name, date, time)
+        layer_dir = os.path.dirname(impact_function.impact.source())
```

Comment:
You shouldn't do that. The source might not be a directory, if the datastore is a database for instance.